crownheightsaid/mutual-aid-app
# When delivery is assigned, remove cross-streets from request post in Slack.

This PR will make the Slack app remove map links from request posts when they're no longer needed.

If the link has alt text (the cross-streets by default), it will leave the alt text in place.
When status is "Delivery Needed":
![maa-pr101-1](https://user-images.githubusercontent.com/1417643/85626999-dcaef280-b63b-11ea-9dd8-dcba9c028e9e.png)

When status is "Delivery Assigned":
![maa-pr101-2](https://user-images.githubusercontent.com/1417643/85627001-df114c80-b63b-11ea-946f-ef68f72ecae8.png)

If there is no alt text, it will simply remove the line.

Testing requires:
- Main Airtable access
- Slack access
- a Slack channel called `#community_needs` (or whatever `REQUESTS_CHANNEL` is set to in `~slack/constants`)
- ngrok running (`npm run local:slack`)
- server running (`npm run local:express`)

Testing process:
1. create a delivery request in "STAGING Intake + Vols: Requests" with the "status" set to "Delivery Needed"
2. post the request in the request channel using the Slack app action
3. in Airtable, update the status to "Delivery Assigned"
4. check the Slack post, the link "Cross Sreets" line should be gone

Also!!! This feature has a test associated with it. To run it, run `npm run test`

closes #89
